src/Catalyst.Core.Modules.Kvm/KvmModule.cs

```diff
@@ -21,32 +21,83 @@
 
 #endregion
 
+using System;
 using Autofac;
+using Autofac.Builder;
+using Autofac.Core;
 using Catalyst.Abstractions.Kvm;
 using Nethermind.Core.Specs;
 using Nethermind.Evm;
 using Nethermind.Logging;
 using Nethermind.Store;
+using Module = Autofac.Module;
 
 namespace Catalyst.Core.Modules.Kvm
 {
     public class KvmModule : Module
     {
         protected override void Load(ContainerBuilder builder)
         {
-            builder.RegisterType<DeltaExecutor>().As<IDeltaExecutor>().SingleInstance();
-            builder.RegisterType<KatVirtualMachine>().As<IKvm>().SingleInstance();
             builder.RegisterType<CatalystSpecProvider>().As<ISpecProvider>();
 
-            builder.RegisterType<StateProvider>().As<IStateProvider>().SingleInstance();
-            builder.RegisterType<StorageProvider>().As<IStorageProvider>().SingleInstance();
-            builder.RegisterType<StateUpdateHashProvider>().As<IStateUpdateHashProvider>().SingleInstance();
+            builder.RegisterType<StateUpdateHashProvider>().As<IStateUpdateHashProvider>().SingleInstance(); 
+            
+            // builder.RegisterInstance(new OneLoggerLogManager(new SimpleConsoleLogger())).As<ILogManager>();
             builder.RegisterInstance(LimboLogs.Instance).As<ILogManager>();
 
-            builder.RegisterType<MemDb>().As<IDb>();               // code db
-            builder.RegisterType<StateDb>().As<ISnapshotableDb>(); // state db
-            builder.RegisterType<EthRpcService>().As<IEthRpcService>().SingleInstance();
+            builder.RegisterInstance(new MemDb()).As<IDb>().SingleInstance();               // code db
+            builder.RegisterInstance(new StateDb()).As<ISnapshotableDb>().SingleInstance(); // state db
+
             builder.RegisterType<StateReader>().As<IStateReader>(); // state db
-        }  
+        }
+    }
+
+    public static class ExecutionRegistrations
+    {
+        /// <summary>
+        /// Registers a custom set of the following components: <see cref="IStateProvider"/>, <see cref="IStorageProvider"/>, <see cref="IKvm"/> and <see cref="IDeltaExecutor"/> for the given <paramref name="registration"/>.
+        /// </summary>
+        /// <param name="registration">The registration to be enhanced.</param>
+        /// <param name="builder">The container builder.</param>
+        /// <returns>The <paramref name="registration"/>.</returns>
+        public static IRegistrationBuilder<TLimit, TReflectionActivatorData, TStyle> WithExecutionParameters<TLimit, TReflectionActivatorData, TStyle>(this IRegistrationBuilder<TLimit, TReflectionActivatorData, TStyle> registration, ContainerBuilder builder)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Split this 258 characters long line (which is greater than 200 authorized).](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4979613)